src/pages/dev/amplifier/chain-integration/introduction.mdx

```diff
@@ -1,63 +1,101 @@
-1. Test in Testnet
-    - Ensure your contracts are working as expected.
-    - Verify with developers that they can send messages to and from your chain.
-1. Productionize your relayer
-    - Ensure your relayer is robust, fault-tolerant, and can handle the load of your chain.
-1. Propose to Mainnet
-    - [Make on-chain proposals](/learn/cosmwasm-governance) on Mainnet to create your CosmWasm smart contracts.
-        1. Proposal to store your CosmWasm code.
-        1. Proposal to instantiate your contracts.
-    - Make an on-chain proposal to add your chain to the Amplifier router.
\ No newline at end of file
+## Integration architecture
+
+To connect your chain to the Axelar network via the Interchain Amplifier, you'll need to build or instantiate a [message relayer](/dev/amplifier/chain-integration/introduction#message-relayer) service and the following four smart contracts:
+
+- [Source chain gateway](/dev/amplifier/chain-integration/introduction#source-chain-gateway-contract)
+- [Axelar gateway](/dev/amplifier/chain-integration/introduction#axelar-gateway-contract)
+- [Voting verifier](/dev/amplifier/chain-integration/introduction#verifier-contract)
+- [Multisig prover](/dev/amplifier/chain-integration/introduction#prover-contract)
+
+### Message relayer
+
+- This is a service built with any tech stack that listens for events on your chain and relays them to the Axelar network.
+- For a detailed implementation, see the [Message Relayer example](https://github.com/axelarnetwork/axelar-examples/tree/main/examples/amplifier) on GitHub.
+
+### Source chain gateway contract
+
+- This is the gateway contract for your source chain, built using the same tech stack as that chain. It serves as the primary API for developers sending messages from your chain.
+- The contract must implement the [Cross-Chain Gateway Protocol](https://github.com/axelarnetwork/cgp-spec).
+- EVM chains can re-use the existing [EVM Gateway](https://github.com/axelarnetwork/axelar-cgp-solidity).
+
+### Axelar gateway contract
+
+- This is a smart contract on the Axelar network that checks the validity of a transaction with the verifier contract and forwards it to the Amplifier router.
+- Most integrators will begin by using or customizing the Axelar [Gateway contract](https://github.com/axelarnetwork/axelar-amplifier/tree/main/contracts/gateway) (current version: 0.2.0).
+- [Instantiation parameters](https://github.com/axelarnetwork/axelar-amplifier/blob/main/contracts/gateway/src/msg.rs) can be found on GitHub.
+
+### Voting verifier contract
+
+- This is a smart contract on the Axelar network that verifies the validity of transactions on your chain.
+- Most integrators will begin by using or customizing the Axelar [Voting Verifier contract](https://github.com/axelarnetwork/axelar-amplifier/tree/main/contracts/voting-verifier) (current version: 0.3.0), which can be customized to your needs for cases such as verification via ZK Proof.
```

Comment:
I don't think listing versions in this doc helps. I think we should specify the versions when we are actually doing the work, not explaining it conceptually.

Reply:
SG. Removed current versions for now. Still looking for the versioning matrix, will update this page when I find it.